### PR TITLE
Added hex color conversion

### PR DESCRIPTION
**Pull request by jsheely**
Solves issue https://github.com/spectreconsole/spectre.console/issues/1354

I'm not sure if this is desired but seemed like an easy quality of life feature to allow hex to rgb for creating colors.

---
Please upvote :+1: this pull request if you are interested in it.

### CODE REVIEW

**Review by patriksvensson — DISMISSED** [review]
* The constructor only tests the happy path.
* I think `public static Color.FromHex(string hex)` would be a more desirable approach here from an API perspective.
* All errors will need to be resolved.

**Comment by nils-a** [comment]
> * I think `public static Color.FromHex(string hex)` would be a more desirable approach here from an API perspective.

I'd vote for a combination of `public static Color ParseHex(string hex)` and `public static bool TryParseHex(string hex, out Color color)`

**Comment by patriksvensson** [comment]
We use "ToHex" so "FromHex" and "TryFromHex" is more consistent API wise

**Comment by FrankRay78** [comment]
It looks like `hex[2..4]` is not supported in netstandard 2.0

**Comment by FrankRay78** [comment]
Unfortunately, the build still breaks @jsheely - see the build log above, looks like simple char/string conversion error. Want to update and push again?

**Comment by FrankRay78** [comment]
Thank you @jsheely, if no one else gets to it first, I'll review (and hopefully merge) your PR this week.

File: test/Spectre.Console.Tests/Unit/ColorTests.cs [hunk 1]
```diff
@@ -4,6 +4,14 @@ public sealed class ColorTests
...
     public sealed class TheEqualsMethod
     {
+        [Fact]
+        public void Initialize_Color_With_Hex()
+        {
+            var name = Color.FromHex("#800080").ToString();
+
+            name.ShouldBe("#800080 (RGB=128,0,128)");
+        }
+
```
[the remark this answers is not in the document]
Comment (FrankRay78) [reply test/Spectre.Console.Tests/Unit/ColorTests.cs:61]: I've implemented this, and pushed directly to the branch.

File: test/Spectre.Console.Tests/Unit/ColorTests.cs [hunk 2]
```diff
@@ -4,6 +4,14 @@ public sealed class ColorTests
 {
     public sealed class TheEqualsMethod
     {
+        [Fact]
+        public void Initialize_Color_With_Hex()
```
[the remark this answers is not in the document]
Comment (FrankRay78) [reply test/Spectre.Console.Tests/Unit/ColorTests.cs]: I've implemented this, and pushed directly to the branch.

**Review by FrankRay78 — APPROVED** [review]

**Review by FrankRay78 — APPROVED** [review]

**Comment by FrankRay78** [comment]
Dear @patriksvensson, I have rebased this branch, implemented full unit test coverage for both happy and unhappy paths, and ensured the failure to parse an invalid colour raises a `FormatException` with (what seems to me) to be a reasonable message. 

I think it's suitable for merging, please re-review when you can.

**Comment by nils-a** [comment]
Do we want to support the short-hands for hex values? (E.g. using `#a1b` instead of `#aa11bb`)

**Comment by FrankRay78** [comment]
> Do we want to support the short-hands for hex values? (E.g. using `#a1b` instead of `#aa11bb`)

You tell me @nils-a 😉, as someone who would like to merge this PR for me...    I'm happy to implement the short color code if you'd like.

This (seemingly trivial) PR has been hanging around for too long for my liking.

**Comment by nils-a** [comment]
> This (seemingly trivial) PR has been hanging around for too long for my liking.

Agreed. Let's use it as it is.

@patriksvensson I feel your requested changes were addressed, right?

**Comment by FrankRay78** [comment]
I'm pretty sure you want to merge this PR @nils-a .  Remember that really nice, warm fuzzy feeling from doing so?   It's waiting.

**Comment by nils-a** [comment]
@jsheely your changes have been merged, thanks for your contribution :+1:

**Pull request merged**